go-testing-tutorial/main_test.go

```diff
@@ -138,3 +139,24 @@ func TestHttp(t *testing.T) {
 		t.Error("Test failed. Expected status to equal expected service response.")
 	}
 }
+
+func TestSearchHandlerShouldReturn404IfNoSearchQueryIsPresent(t *testing.T) {
+	req, err := http.NewRequest("GET", "/", nil)
+	if err != nil {
+		return
+	}
+	w := httptest.NewRecorder()
+	searchHandler(w, req)
+
+	resp := w.Result()
+	body, _ := ioutil.ReadAll(resp.Body)
+
+	if resp.StatusCode != 400 {
+		t.Error("Wrong status code")
+	}
+	fmt.Println(string(body))
+
+	if string(body) != "{\"status\": \"expected service response\"}{\"jsonData\": \"{\"Results\":[\"Cutie\",\"Autism\",\"iPhone 12\"]}\"}" {
```

Comment:
I think the response body should only be `"{\"Results\":[\"Cutie\",\"Autism\",\"iPhone 12\"]}`, no need to wrap this with `jsonData` and we dont need the additional json object that has the `status` in there.